lib/extensions.js

```diff
@@ -181,9 +181,11 @@ function getBinaryName() {
   } else if (pkg.nodeSassConfig && pkg.nodeSassConfig.binaryName) {
     binaryName = pkg.nodeSassConfig.binaryName;
   } else {
+    var jsEngine = process.jsEngine ? '-' + process.jsEngine : '';
     binaryName = [process.platform, '-',
                   process.arch, '-',
-                  process.versions.modules].join('');
+                  process.versions.modules,
+                  jsEngine].join('');
```

Comment:
Prefer `_` if `val` is just a pass through.

```js
.filter(function(_) { return _; });
```

Personally I prefer code to be explicit.

```js 
var binaryName = [
  process.platform,
  process.arch,
  process.versions.modules
];

if (process.jsEngine) {
  binaryName.push(process.jsEngine);
}

return binaryName.join('-');
```